xmtp_api_d14n/src/queries/client_bundle.rs

```diff
@@ -174,12 +189,23 @@ impl ClientBundleBuilder {
 
             let gateway_client = gateway_client_builder.build()?;
             let multi_node = multi_node.build()?;
-            let rw = ReadWriteClient::builder()
-                .read(multi_node)
-                .write(gateway_client)
-                .filter(PAYER_WRITE_FILTER)
-                .build()?;
-            let client = rw.arced();
+
+            let client = if auth_callback.is_some() || auth_handle.is_some() {
+                let auth = crate::AuthMiddleware::new(gateway_client, auth_callback, auth_handle);
+                ReadWriteClient::builder()
```

Comment:
you can seperate out the common parts of the builder so that it's more obvious what the auth is switching here, i.e

```rust
let mut rw = ReadWriteClient::builder()
    .read(multi_node)
    .filter(PAYER_WRITE_FILTER);

if (condition) {
   rw.write(auth);
} else {
   rw.write(gateway);
}

let rw = rw.build()?;
```